File: weather_script.js
const apiKey = '70404f6b22c58e5f19cae492d098d4e9'; // Replace with your OpenWeatherMap API key
const apiUrl = 'http://api.openweathermap.org/data/2.5/weather%22';

async function fetchWeatherData(city) {
    const url = `${apiUrl}?q=${city}&appid=${apiKey}&units=metric`;

    try {
        const response = await fetch(url);
        if (!response.ok) {
            const errorData = await response.json();
            throw new Error(`Network response was not ok: ${errorData.message}`);
        }
        const data = await response.json();
        displayWeatherData(data);
    } catch (error) {
        console.error('Fetch error:', error);
    }
}

function displayWeatherData(data) {
    const temperature = data.main.temp;
    const humidity = data.main.humidity;
    const weatherCondition = data.weather[0].description;

    console.log(`Temperature: ${temperature}°C`);
    console.log(`Humidity: ${humidity}%`);
    console.log(`Weather Condition: ${weatherCondition}`);
}

// Example usage
fetchWeatherData('London');

File: weather_script.py
import requests

import requests

def get_weather(city_name, api_key):
    base_url = "http://api.openweathermap.org/data/2.5/weather"
    params = {
        'q': city_name,
        'appid': api_key,
        'units': 'metric'  # Use 'imperial' for Fahrenheit
    }
    response = requests.get(base_url, params=params)
    if response.status_code == 200:
        data = response.json()
        return data
    else:
        return None

if __name__ == "__main__":
    city_name = input("Enter city name: ")
    api_key = "d8e693057635f8f045828226aa9ab1ba"  # Replace with your actual API key
    weather_data = get_weather(city_name, api_key)
    if weather_data:
        print(f"Weather in {city_name}:")
        print(f"Temperature: {weather_data['main']['temp']}°C")
        print(f"Weather: {weather_data['weather'][0]['description']}")
    else:
        print("Failed to get weather data")